include/neml2/models/Model.h

```diff
@@ -165,14 +171,14 @@ class Model : public NEML2Object, public LabeledAxisInterface, public NonlinearS
   is added.
 
   NOTE: We also register this model as a submodule (in torch's language), so that when *this*
-  `Model` is send to another device, the registered `Model` is also sent to that device.
+  `Model` is sent to another device, the registered `Model` is also sent to that device.
   */
   void register_model(std::shared_ptr<Model> model, bool merge_input = true);
 
   virtual void set_residual(BatchTensor<1> x, BatchTensor<1> r, BatchTensor<1> * J = nullptr) const;
 
   /// Models *this* model may use during its evaluation
-  std::vector<std::shared_ptr<Model>> _registered_models;
+  std::vector<Model *> _registered_models;
```

Comment:
Why did you make this change?  Is there any risk that the models get dellocated before they are used here?

Reply:
Fine, I assume git blame will show it as your fault if we run into trouble in the future :)